sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/EliminateSortsSuite.scala

```diff
@@ -310,4 +310,109 @@ class EliminateSortsSuite extends PlanTest {
     val correctAnswer = PushDownOptimizer.execute(noOrderByPlan.analyze)
     comparePlans(optimized, correctAnswer)
   }
+
+  testRepartitionOptimization(
```

Comment:
I prefer not to replace `test` because it makes tests difficult to run individually (at least in my IntelliJ environment).

It also tends to increase readability. Here, you're passing a function to `testRepartitionOptimization` that gets passed a function that modifies the logical plan. I think it would be easier to read if these were separate suites, with a suite-level repartition function:

```scala
class EliminateSortsInRepartitionSuite extends ... {
  def repartition(plan: LogicalPlan): LogicalPlan = plan.repartition(10)

  test("remove sortBy") {
    val plan = testRelation.select('a, 'b).sortBy('a.asc, 'b.desc)
    val planWithRepartition = repartition(plan)
    ...
  }
}

class EliminateSortsInRepartitionByExpressionSuite extends EliminateSortsInRepartitionSuite {
  override def repartition(plan: LogicalPlan): LogicalPlan = plan.distribute('a, 'b)(10)
}
```

Reply:
I think this pattern is common for the codebase but I agree having separate suites makes more sense here. Updated.